### ISSUE TITLE
Addon-a11y: deterministically create config object

### ISSUE THREAD

**Issue by donaldpipowitch** [storybookjs/storybook#8832]
**Is your feature request related to a problem? Please describe.**
The problem was described in more detail here: https://github.com/storybookjs/storybook/pull/8779#discussion_r345121193

There are currently four levels which need to be considered in order to deterministically create the config object for `a11y`. 

1. `addon-a11y` has a default config object (which is also slightly changed at some point, but I don't know where?)
2. we have global-level custom config (import { addParameters } from '@storybook/react'; addParameters({ a11y })),
3. we have component-level custom config (export default { parameters: { a11y } } or storiesOf('...', module).addParameters({ a11y })),
4. we have story-level custom config (Example.story = { parameters: { a11y } }; or storiesOf('...', module).add('...', () => {}, { a11y })).

For every `run` of  `addon-a11y` we need to be able to deterministically get the correct config object by merging everything from level 1 to 4.

⚠️ Note: I haven't created an addon so far. I'm not sure if this will be a fix specific for `addon-a11y` or if this needs to be supported upstream by Storybook itself.

**Describe the solution you'd like**

Before the `run` function is called, there needs to a function call like `getConfig` or something like that which correctly merges all possible configurations from every level described above.

**Describe alternatives you've considered**

I'm not sure if this should be a more general feature offered by the Storybook core for every addon?

**Are you able to assist bring the feature to reality?**

Currently not. I'd need to know at least two things:
1. Where is the default config object (mentioned as level 1) altered?
2. Should this be a fix specific to this addon or more general?

**Additional context**

Related MR which has a _limited_ workaround for this problem: https://github.com/storybookjs/storybook/pull/8779
